captum/optim/_param/image/images.py

```diff
@@ -428,11 +425,13 @@ def __init__(
         decorrelate_init: bool = True,
     ) -> None:
         super().__init__()
-        self.decorrelate = decorrelation_module
+        self.decorrelate = (
+            decorrelation_module.cpu() if decorrelation_module is not None else None
+        )
         if init is not None:
+            assert not init.is_cuda
```

Comment:
Enforcing the tensor to be on CPU without warning why, will make the user to think that there is a bug in the code.
We should explicitly put is on the CPU. I assume this is because of `decorrelation_module.cpu()`

I think that it would be good to print a warning message why we put `decorrelation_module` on the CPU and explain it in the documentation.

Perhaps also adding a TODO so that we can make the module more flexible and don't have to move to CPU device.

Reply:
The init tensor had to be put on the CPU because `FFTImage` expected it to be on the CPU, but I just fixed it so that `spectrum_scale` is now placed on to the init tensor's device.

In order to avoid any future issues with the decorrelation module being created in the function signature, I added a deepcopy line that only runs when it's a `ToRGB` instance.

I also removed the device assertion and made `ToRGB` place the transform on the input tensor's device as well, so that it becomes a no-op when the buffer is placed on the target device. 

The device issues should now be resolved I think?

Reply:
I removed the deepcopy line as it was redundant.

Reply:
It looks cleaner now.

Here squash_func is not assigned, right? I wonder why we use different `squash_func`s depending on whether `init` is provided or not.
```
if squash_func is None: 
 def squash_func(x: torch.Tensor) -> torch.Tensor: 
   return x.clamp(0, 1)
```

Reply:
@NarineK The different squash function for when init tensors are provided was something I found to produce better results with images.

The TODO line is because named dimensions are not fully supported by PyTorch yet.

Reply:
interesting, why are inner functions better in this case ?
`rename` is used in line 440 too. I we can leave it but ideally we should use it for the PT versions that support it.